weatherman.py

```diff
@@ -0,0 +1,61 @@
+import argparse
+from code_files.weather_parser import WeatherParser
+from code_files.weather_reporter import WeatherReporter
+from code_files.report_calculator import ReportCalculator
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("extract_to", type=str)
+    parser.add_argument("-e", "--extremes", type=int)
+    parser.add_argument("-a", "--averages", type=str)
+    parser.add_argument("-c", "--chart", type=str)
+    parser.add_argument("--inline", action="store_true")
+
+    args = parser.parse_args()
+
+    parser = WeatherParser(args.extract_to)
+    parser.extract_zip()
+    parser.parse_all_files()
+
+    if args.extremes:
+        yearly_readings = [
+            r for r in parser.weather_readings if r.date.startswith(str(args.extremes))
+        ]
+        if yearly_readings:
+            weather_extremes = ReportCalculator.compute_extreme_stats(yearly_readings)
+            WeatherReporter.generate_report_extremes(weather_extremes)
```

Comment:
We are using `ReportCalculator` & `WeatherReporter` classes as a functions here. Let's make sure we are following oop approach.

Reply:
[Refactoring code to make use of OOP for `ReportCalculator` & `WeatherReporter` classes](https://github.com/ttqureshi/WeatherMan/pull/1/commits/4b32012e0123a6b56cffad28fac085e86fb58a6e)